clang-tools-extra/clang-doc/HTMLMustacheGenerator.cpp

```diff
@@ -0,0 +1,523 @@
+  Object Obj = Object();
+  std::string EnumType = I.Scoped ? "enum class " : "enum ";
+  EnumType += I.Name;
+  bool HasComment = std::any_of(
+      I.Members.begin(), I.Members.end(),
+      [](const EnumValueInfo &M) { return !M.Description.empty(); });
+  Obj.insert({"EnumName", EnumType});
+  Obj.insert({"HasComment", HasComment});
+  Obj.insert({"ID", llvm::toHex(llvm::toStringRef(I.USR))});
+  Value Arr = Array();
+  for (const EnumValueInfo& M: I.Members) {
+    Value EnumValue = Object();
+    EnumValue.getAsObject()->insert({"Name", M.Name});
+    if (!M.ValueExpr.empty())
+      EnumValue.getAsObject()->insert({"ValueExpr", M.ValueExpr});
+    else
+      EnumValue.getAsObject()->insert({"Value", M.Value});
+    
+    if (!M.Description.empty()) {
+      Value ArrDesc = Array();
+      for (const CommentInfo& Child : M.Description) 
+        ArrDesc.getAsArray()->emplace_back(extractValue(Child));
+      EnumValue.getAsObject()->insert({"EnumValueComments", ArrDesc});
+    }
+    Arr.getAsArray()->emplace_back(EnumValue);
+  }
+  Obj.insert({"EnumValues", Arr});
+  
+  if (!I.Description.empty()) {
+    Value ArrDesc = Array();
+    for (const CommentInfo& Child : I.Description) 
+      ArrDesc.getAsArray()->emplace_back(extractValue(Child));
+    Obj.insert({"EnumComments", ArrDesc});
+  }
+  
+  if (I.DefLoc.has_value()) {
+    Location L = *I.DefLoc;
+    if (CDCtx.RepositoryUrl.has_value())
+      Obj.insert({"Location", extractValue(L,
+                                           StringRef{*CDCtx.RepositoryUrl})});
+    else
+      Obj.insert({"Location", extractValue(L)});  
+  }
+  
+  return Obj;
+}
+
+void extractScopeChildren(const ScopeChildren &S, Object &Obj, 
+                          StringRef ParentInfoDir,
+                          const ClangDocContext &CDCtx) {
+  Value ArrNamespace = Array();
+  for (const Reference& Child : S.Namespaces)
+    ArrNamespace.getAsArray()->emplace_back(extractValue(Child, ParentInfoDir));
+  
+  if (!ArrNamespace.getAsArray()->empty())
+    Obj.insert({"Namespace", Object{{"Links", ArrNamespace}}});
+  
+  Value ArrRecord = Array();
+  for (const Reference& Child : S.Records)
+    ArrRecord.getAsArray()->emplace_back(extractValue(Child, ParentInfoDir));
+  
+  if (!ArrRecord.getAsArray()->empty())
+    Obj.insert({"Record", Object{{"Links", ArrRecord}}});
+  
+  Value ArrFunction = Array();
+  Value PublicFunction = Array();
+  Value ProtectedFunction = Array();
+  Value PrivateFunction = Array();
+  
+  for (const FunctionInfo& Child : S.Functions) {
+    Value F = extractValue(Child, ParentInfoDir, CDCtx);
+    AccessSpecifier Access = Child.Access;
+    if (Access == AccessSpecifier::AS_public)
+      PublicFunction.getAsArray()->emplace_back(F);
+    else if (Access == AccessSpecifier::AS_protected)
+      ProtectedFunction.getAsArray()->emplace_back(F);
+    else
+      ArrFunction.getAsArray()->emplace_back(F);
```

Comment:
I see you using `ArrFunction.getAsArray()` a lot. We should probably cache a reference to it.